src/convertors/logseq/logseq-convertor.ts

```diff
@@ -0,0 +1,136 @@
+import {markdownToHtml} from 'helpers/markdown'
+import {validateNotes} from 'helpers/validate'
+
+import {tryParseDate} from './logseq-helpers'
+import {LogseqBlock, LogseqConversionError, LogseqNote} from './types'
+import {
+  Backlink,
+  ConvertedNote,
+  ConvertOptions,
+  Convertor,
+  ConvertResponse,
+  REFLECT_HOSTNAME,
+} from '../../types'
+
+export class LogseqConvertor implements Convertor {
+  graphId: string
+  linkHost: string
+  noteIds: Record<string, string> = {}
+
+  constructor({
+    graphId,
+    linkHost = REFLECT_HOSTNAME,
+  }: {
+    graphId: string
+    linkHost?: string
+  }) {
+    this.graphId = graphId
+    this.linkHost = linkHost
+  }
+
+  accept = {'application/json': ['.json']}
+
+  convert({data}: ConvertOptions): ConvertResponse {
+    const parsed = JSON.parse(data)
+
+    if (parsed.version !== 1) {
+      throw new LogseqConversionError('Only able to convert Logseq file version 1')
+    }
+
+    const notes = parsed.blocks as LogseqNote[]
```

Comment:
You might want to coerce the whole object further up rather than just the blocks.